File: mobile/lib/shared/models/payment.dart
class Payment {
  const Payment({
    required this.id,
    required this.reference,
    this.invoiceId,
    required this.amount,
    required this.method,
    required this.date,
    required this.status,
  });

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
        id: json['id'] as int,
        reference: json['reference'] as String? ?? '',
        invoiceId: json['invoiceId'] as int?,
        amount: (json['amount'] as num?)?.toInt() ?? 0,
        method: json['method'] as String? ?? '',
        date: json['date'] as String? ?? '',
        status: json['status'] as String? ?? 'Pending',
      );

  final int id;
  final String reference;
  final int? invoiceId;
  final int amount;
  final String method;
  final String date;
  final String status;
}

File: backend/app/Models/Payment.php
<?php

namespace App\Models;

use App\Models\Concerns\BelongsToTenant;
use Database\Factories\PaymentFactory;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

#[Fillable(['member_id', 'invoice_id', 'amount', 'method', 'date', 'status'])]
class Payment extends Model
{
    /** @use HasFactory<PaymentFactory> */
    use BelongsToTenant, HasFactory;

    protected function casts(): array
    {
        return [
            'date' => 'date',
        ];
    }

    public function member(): BelongsTo
    {
        return $this->belongsTo(Member::class);
    }

    public function invoice(): BelongsTo
    {
        return $this->belongsTo(Invoice::class);
    }
}
